tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,445 @@
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "CNN Newsroom With Fredricka Whitfield"
+}
+
+RECORDING = {
+    "callsign": "CNNHD",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": True,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "CNN Newsroom With Fredricka Whitfield",
+    'uniqueId': '12345',
+    'episodeTitle': 'CNN Recorded'
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class mockDIRECTVClass:
+    """A fake DirecTV DVR device."""
+
+    def __init__(self, ip, port=8080, clientAddr='0'):
+        """Initialize the fake DirecTV device."""
+        self._host = ip
+        self._port = port
+        self._device = clientAddr
+        self._standby = True
+        self._play = False
+
+        _LOGGER.debug("INIT with host: %s, port: %s, device: %s",
+                      self._host, self._port, self._device)
+
+        self._locations = LOCATIONS
+
+        self.attributes = LIVE
+
+    def get_locations(self):
+        """Mock for get_locations method."""
+        _LOGGER.debug("get_locations called")
+        test_locations = {
+            'locations': self._locations,
+            'status': {
+                'code': 200,
+                'commandResult': 0,
+                'msg': 'OK.',
+                'query': '/info/getLocations'
+            }
+        }
+
+        return test_locations
+
+    def get_standby(self):
+        """Mock for get_standby method."""
+        _LOGGER.debug("STANDBY is: %s", self._standby)
+        return self._standby
+
+    def get_tuned(self):
+        """Mock for get_tuned method."""
+        _LOGGER.debug("get_tuned called")
+        if self._play:
+            self.attributes['offset'] = self.attributes['offset']+1
+
+        test_attributes = self.attributes
+        test_attributes['status'] = {
+            "code": 200,
+            "commandResult": 0,
+            "msg": "OK.",
+            "query": "/tv/getTuned"
+        }
+        return test_attributes
+
+    def key_press(self, keypress):
+        """Mock for key_press method."""
+        _LOGGER.debug("Key Press: %s", keypress)
+        if keypress == 'poweron':
+            self._standby = False
+            self._play = True
+        elif keypress == 'poweroff':
+            self._standby = True
+            self._play = False
+        elif keypress == 'play':
+            self._play = True
+        elif keypress == 'pause' or keypress == 'stop':
+            self._play = False
+
+    def tune_channel(self, source):
+        """Mock for tune_channel method."""
+        _LOGGER.debug("Change channel %s", source)
+        self.attributes['major'] = int(source)
+
+
+class TestSetupDirectvMediaPlayer(unittest.TestCase):
```

Comment:
We prefer standalone pytest test functions for new tests. See eg the deconz test suite.

Reply:
@MartinHjelmare , all changed. Let me know. :-)